lib/views/pages/post/post_page_widget/bottom_bar_add_comment.dart

```diff
@@ -1,59 +1,105 @@
 import "package:flutter/material.dart";
+import "package:flutter_hooks/flutter_hooks.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/utils/ui/circular_value.dart";
 import "package:proxima/utils/ui/user_avatar.dart";
+import "package:proxima/viewmodels/comment_view_model.dart";
+import "package:proxima/viewmodels/new_comment_view_model.dart";
 
-class BottomBarAddComment extends StatelessWidget {
+class BottomBarAddComment extends HookConsumerWidget {
   static const commentUserAvatarKey = Key("commentUserAvatar");
   static const addCommentTextFieldKey = Key("addCommentTextField");
   static const postCommentButtonKey = Key("postCommentButton");
 
   static const _textFieldHintAddComment = "Add a comment";
 
   final String currentDisplayName;
+  final PostIdFirestore parentPostId;
 
   const BottomBarAddComment({
     super.key,
+    required this.parentPostId,
     required this.currentDisplayName,
   });
 
   @override
-  Widget build(BuildContext context) {
-    return Row(
-      // Align items to the start of the cross axis
-      crossAxisAlignment: CrossAxisAlignment.start,
-      children: [
-        Padding(
+  Widget build(BuildContext context, WidgetRef ref) {
+    final contentController = useTextEditingController();
+
+    final asyncNewCommentState = ref.watch(
+      newCommentStateProvider(parentPostId),
+    );
+
+    return CircularValue(
+      value: asyncNewCommentState,
+      builder: (context, newCommentState) {
+        // The avatar of the current user on the left
```

Comment:
It would make sense to create an additional `StatelessWidget` to abstract the content of the circular value here. The builder method is way too long and hard to read imo.
We could abstract it in two or three widgets that we could place in `views/pages/post/components`.

Reply:
You are right, the circular value is not very readable.
I'll do my best to provide a good refactor.
So, I have created 3 new widgets to separate the UI elements of the bottom bar.
And I have putted them under `views/pages/post/components/comment`

I do that in b948c795a3e24729d2311df6b1ea1b05c05c490b.